### PR TITLE
chore: refactor module

### PR DESCRIPTION
**Pull request by userquin**
### 🔗 Linked issue

This PR just moves logic from `module.ts` to the corresponding module: for example, when configuring nitro, the current `module.ts` will build the `AdditionalSetupNitroParams` (being used or not), with the new approach, the `nitro.ts` module will build the `AdditionalSetupNitroParams` when required.

The idea is to have the `module.ts` to orchestrate the I18n configuration: any shared data should be on the context, and the modules should do their work. The `module.ts` will call the initialization sequence correctly, and modules preparing shared data should add their stuff to the context (check `prepare/locale-info.ts` and `prepare/runtime.ts`).

Any module should only accept the context and nuxt as parameters: using destructuring on the context to get used data.

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by BobbieGoede** [comment]
approving to run ci

File: src/prepare/runtime.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      normalizedLocales,
+      dev,
+      isSSG,
+      parallelPlugin: options.parallelPlugin
+    })
+  }
+
+  ctx.genTemplate = genTemplate
+
+  nuxt.options.runtimeConfig.public.i18n.locales = simplifyLocaleOptions(nuxt, defu({}, options))
```
Comment (BobbieGoede) [inline src/prepare/runtime.ts:44]: I'm guessing we can't move this to `prepare/runtime-config.ts` right? 🤔

Comment (userquin) [reply src/prepare/runtime.ts:44]: oh, we can do what we want here

**Review by BobbieGoede — APPROVED** [review]

File: src/prepare/runtime.ts (hunk 1 above)
Comment (userquin) [reply src/prepare/runtime.ts:44]: we can move runtime config after runtime and then we can move it: do you want me to move it?

**Review by kazupon — APPROVED** [review]
This PR is amazing!
Thanks!

**Pull request merged**